yt/config.py

```diff
@@ -196,18 +165,52 @@ def write(self, file_handler):
             with open(file_handler, mode="w") as fh:
                 fh.write(config_as_str)
 
+    @staticmethod
+    def get_global_config_file():
+        return os.path.join(CONFIG_DIR, "yt.toml")
+
+    @staticmethod
+    def get_local_config_file():
+        return os.path.join(os.path.abspath(os.curdir), "yt.toml")
+
 
-# Walk the tree up until we find a config file
+OLD_CONFIG_FILE = os.path.join(CONFIG_DIR, "ytrc")
+_global_config_file = YTConfig.get_global_config_file()
+_local_config_file = YTConfig.get_local_config_file()
+
+if os.path.exists(OLD_CONFIG_FILE):
+    if os.path.exists(_global_config_file):
+        msg = (
+            f"The configuration file {OLD_CONFIG_FILE} is deprecated in "
+            f"favor of {_global_config_file}. Currently, both are present. "
+            "Please manually remove the deprecated one to silence "
+            "this warning."
+        )
+        warnings.warn(msg)
+    else:
+        msg = (
+            f"The configuration file {OLD_CONFIG_FILE} is deprecated. "
+            f"Please migrate your config to {_global_config_file} by running: "
+            "'yt config migrate'"
+        )
+        raise SystemExit(msg)
+
+
+if not os.path.exists(_global_config_file):
+    cfg = {"yt": {}}
+    try:
+        with open(_global_config_file, mode="w") as fd:
+            toml.dump(cfg, fd)
+    except OSError:
+        warnings.warn("unable to write new config file")
```

Comment:
Hum, I don't really like this. I assume that this block was initially written to ease development, but I don't see a good reason to keep it. This is not directly related to this PR however.

Reply:
The issue is that if `~/.config/yt/yt.toml` is not writeable, you cannot use yt at all...

Reply:
I get that this is currently the case. My point is that it shouldn't :)

Reply:
The thing is that yt's configuration is optional, so I don't think we should force the user to have one, should we?

Reply:
I just realised that I misread the code when I first commented. In the end we agree, please disregard.